src/tests/base_test.py

```diff
@@ -826,16 +826,18 @@ def set_iptables_rules_for_nodes(
             cluster: Cluster,
             iptables_rules: List[IptableRule],
             given_nodes=None,
+            start_stop_nodes=True,
         ):
             given_node_ips = []
             given_nodes = given_nodes or cluster.nodes.nodes
-
-            if cluster.enable_image_download:
-                cluster.generate_and_download_infra_env(iso_download_path=cluster.iso_download_path)
-            cluster.nodes.start_given(given_nodes)
+            if start_stop_nodes:
+                if cluster.enable_image_download:
+                    cluster.generate_and_download_infra_env(iso_download_path=cluster.iso_download_path)
+                cluster.nodes.start_given(given_nodes)
             for node in given_nodes:
                 given_node_ips.append(node.ips[0])
-            cluster.nodes.shutdown_given(given_nodes)
+            if start_stop_nodes:
+                cluster.nodes.shutdown_given(given_nodes)
```

Comment:
we have two identical if statements 
L833  & L839

Maybe you meant 
```suggestion
            if not start_stop_nodes:
                cluster.nodes.shutdown_given(given_nodes)
```

it will be better this way 

```suggestion
            if start_stop_nodes:
                if cluster.enable_image_download:
                    cluster.generate_and_download_infra_env(iso_download_path=cluster.iso_download_path)
                cluster.nodes.start_given(given_nodes)
            else:
               cluster.nodes.shutdown_given(given_nodes)
            for node in given_nodes:
                given_node_ips.append(node.ips[0])
                
```

Reply:
ok , let me change the logic a little bit .  basically shutdown should not be here ...